Marlin/Configuration_adv.h

```diff
@@ -662,6 +662,24 @@
    */
   //#define AUTO_REPORT_SD_STATUS
 
+  /**
+   * If using a bootloader that supports SD-Firmware-Flashing, add
+   * an extra menu item to activate SD-Flashing on reboot.
+   * 
+   * Requires ATMEGA2560 (Arduino Mega)
+   * 
+   * Tested with this bootloader:
+   *   https://github.com/FleetProbe/MicroBridge-Arduino-ATMega2560
+   */
+  #ifdef __AVR_ATmega2560__
+    //#define SD_FIRMWARE_UPGRADE
+    #if ENABLED(SD_FIRMWARE_UPGRADE)
+      #define SD_FIRMWARE_UPGRADE_EEPROM_ADDRESS        0x1FF
```

Comment:
There's a good chance that EEPROM settings could occupy more than 511 bytes. Is this value configurable in the bootloader?

Reply:
No, unfortunately this is a fixed address by the bootloader. You can have a look at their Github. At startup, the bootloader will look at the value at this address. If it is 0xF0, the bootloader will automatically search for a firmware.bin on a connected SD Card. If both is the case, it will flash the firmware.